catalog-scripts/idp-catalog-wizard-bitbucket.py

```diff
@@ -0,0 +1,175 @@
+import argparse
+import requests
+import os
+from requests.auth import HTTPBasicAuth
+import argparse
+from requests.packages.urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
+import subprocess
+
+current_directory = os.path.basename(os.getcwd())
+branch  = "main"
+
+yaml_content_template = """
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  tags:
+    - auto-generated
+    - auto
+  name: <repo_name>
+  annotations:
+    backstage.io/source-location: url:<repo_path>
+spec:
+  type: service
+  domian: <workspace_name>
+  system: <project>
+  service: <repo_name>
+  lifecycle: experimental
+  owner: Harness_Account_All_Users
+"""
+
+prefix_path = "services"
+page_size = 50
+def list_repositories(workspace, project_key, username, app_password, repo_pattern=None):
+        
+    page = 1
+    names = []
+    if(project_key == None):
+        url = f"https://api.bitbucket.org/2.0/repositories/{workspace}"
+    else:
+        url = f"https://api.bitbucket.org/2.0/repositories/{workspace}?q=project.key%3D%22{project_key}%22"
+
+    while True:
+
+        if(project_key == None):
+            paginated_url = f'{url}?page={page}&pagelen={page_size}'
+        else:
+            paginated_url = f'{url}&page={page}&pagelen={page_size}'
+
+        response = requests.get(paginated_url, auth=HTTPBasicAuth(username, app_password))
+        if response.status_code == 200:
+            repositories = response.json()['values']
+
+            for entry in repositories:
+                if entry['name'] != current_directory:
```

Comment:
Does BB allow the same repo name in 2 different projects? If yes, how will this script handle it?

Reply:
Verified this. Repo name is unique in a given workspace